cli/sql.go

```diff
@@ -271,6 +272,15 @@ func runTerm(cmd *cobra.Command, args []string) error {
 	if len(cliContext.execStmts) > 0 {
 		// Single-line sql; run as simple as possible, without noise on stdout.
 		return runStatements(conn, cliContext.execStmts)
+	} else if !isatty.IsTerminal(os.Stdin.Fd()) {
```

Comment:
will this cause invocations as in #5981 to lose their buffering? If so, :-1:. I don't want my 10gb data dump to be held in memory before executing.